src/runtime/panic.go

```diff
@@ -53,11 +65,16 @@ func _panic(message interface{}) {
 		frame := (*deferFrame)(task.Current().DeferFrame)
 		if frame != nil {
 			frame.PanicValue = message
-			frame.Panicking = true
+			frame.Panicking = panicking
 			tinygo_longjmp(frame)
 			// unreachable
 		}
 	}
+	if panicking == panicGoexit {
+		// Call to Goexit() instead of a panic.
+		// Exit the goroutine instead of printing a panic message.
+		deadlock()
```

Comment:
I'm not sure `deadlock` is actually the right thing here.   This means (for example) if we spawn 1000 goroutines and  `Goexit()` all of them, they'll still be hanging around in the scheduler.  How do we get them to *actually* clean up?  What does a "regular" goroutine do at the end of its lifecycle?

Reply:
`deadlock` needs to be changed, yes. But at the moment, it will do the exact same thing a goroutine does when it returns: it calls `Pause` without a way to wake it up. At that point the GC should be able to clean them up (assuming no false positives and such).

I plan to change this to something better because with threading support, the OS thread really does need to be exited.

Reply:
OK, so maybe for me it's just the name of the routine "deadlock" that didn't communicate what was actually happening.